### ISSUE TITLE
Consecutive call to  Ort::Session::Run() crashes

### ISSUE THREAD

**Issue by cerulij** [microsoft/onnxruntime#13476]
### Describe the issue

Hey everybody, I'm using ORT with the C++ API to run a CPU inference on a model. I implemented a class that creates an inference session in its constructor taking a static Ort::Env env and default Ort::SessionOptions. It also has a method that calls Ort::Sesion Run() on the session.

I'm testing the code using GoogleTest for C++. I can create this inference object, Run() the session once, and get the right results. But, when it goes to the next test and tries to create the object and Run() the session, it crashes.

gdb tells me that the issue is in the Ort::Session constructor(env, model, session options); more precisely, some problem with malloc(). But, the issue shows up when I try to run the session; I can create as many objects as I want but when I try to run it, I can only do it once; even across different test files.

I tried making env static because I thought env should be like a singleton but that didn't fix the issue. I also don't think there's a data race when trying to open the model (which is the only variable that's shared across inference objects) because I tried creating a new inference object with a different model and also crashed.

Please let me know if you need more details.

### To reproduce

In C++:
1. Create an Inference class
2. The constructor of the class creates the session (static env, model, default session options)
3. A method of the class calls Session.Run()

Test with Google Test for C++:
1. Make a test to execute an inference using the Inference class
2. Make a second test to do the same
3. The second test should crash

### Platform

Linux

### OS Version

Custom

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.12.1

### ONNX Runtime API

C++

### Architecture

X64

### Execution Provider

Default CPU

**Comment by pranavsharma** [comment]
Can you attach your code? Why are you creating multiple sessions? A session needs to be created only once in the lifecycle of the process.

**Comment by cerulij** [comment]
> Can you attach your code? Why are you creating multiple sessions? A session needs to be created only once in the lifecycle of the process.

I'm creating multiple sessions because I need to run different models; is there any other way to achieve that?

**Comment by pranavsharma** [comment]
Yes, you need a separate session for each model. Can you share the code?
